GVFS/GVFS.FunctionalTests/Tests/GitCommands/GitRepoTests.cs

```diff
@@ -23,13 +25,27 @@ public abstract class GitRepoTests
         protected const string DeepDirectoryWithOneDifferentFile = "FunctionalTests/20181010_DeepFolderOneDifferentFile";
 
         private bool enlistmentPerTest;
+        private bool validateWorkingTree;
 
-        public GitRepoTests(bool enlistmentPerTest)
+        public GitRepoTests(bool enlistmentPerTest, ValidateWorkingTreeOptions validateWorkingTree)
         {
             this.enlistmentPerTest = enlistmentPerTest;
+            this.validateWorkingTree = validateWorkingTree == ValidateWorkingTreeOptions.ValidateWorkingTree;
             this.FileSystem = new SystemIORunner();
         }
 
+        public enum ValidateWorkingTreeOptions

```

Comment:
What is the enum buying us? Can we simplify the pattern even further with the existing `AllBools` data source?

Reply:
By that argument, we should drop the `AllBools` data source and use enums everywhere instead :-). I'm not opposed to doing it either way, but I'd like to simplify our usage of this NUnit API if we can.

Reply:
I think I might be putting too much value into the pretty printing.  

My guess is that when tests fail we'll be most interested in the specific check that failed (and not care as much whether or not we validated the working tree during setup).

Now I'm thinking I'll take this out and stick with `AllBools` for simplicity.